File: server/src/main/java/xyz/chunkstories/server/commands/admin/SayCommand.java
//
// This file is a part of the Chunk Stories Implementation codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.server.commands.admin;

import xyz.chunkstories.api.plugin.commands.Command;
import xyz.chunkstories.api.plugin.commands.CommandEmitter;
import xyz.chunkstories.api.server.Server;
import xyz.chunkstories.server.commands.ServerCommandBasic;

public class SayCommand extends ServerCommandBasic {

	public SayCommand(Server serverConsole) {
		super(serverConsole);
		server.getPluginManager().registerCommand("say", this);
	}

	@Override
	public boolean handleCommand(CommandEmitter emitter, Command command, String[] arguments) {
		if (command.getName().equals("say") && emitter.hasPermission("server.admin")) {
			String message = "";
			for (String a : arguments) {
				message += a + " ";
			}
			server.broadcastMessage("#FFFF00SERVER: " + message);
		}
		return false;
	}

}

File: server/src/main/java/xyz/chunkstories/server/commands/admin/SayCommand.kt
//
// This file is a part of the Chunk Stories Implementation codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.server.commands.admin

import xyz.chunkstories.api.plugin.commands.Command
import xyz.chunkstories.api.plugin.commands.CommandEmitter
import xyz.chunkstories.api.server.Host
import xyz.chunkstories.server.commands.AbstractHostCommandHandler

class SayCommand(serverConsole: Host) : AbstractHostCommandHandler(serverConsole) {

    init {
        host.pluginManager.registerCommand("say", this)
    }

    override fun handleCommand(emitter: CommandEmitter, command: Command, arguments: Array<String>): Boolean {
        if (command.name == "say" && emitter.hasPermission("server.admin")) {
            var message = ""
            for (a in arguments) {
                message += "$a "
            }
            host.broadcastMessage("#FFFF00SERVER: $message")
        }
        return false
    }

}
